string path
string filename
---
bool success
